grpc-proto/proto/bridge-query.proto

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+
+option java_package = "io.stargate.proto";
+option go_package = "stargate/proto";
+
+package stargate;
+
+import "query.proto";
+
+// The response to a Query or Batch message, with additional enriched state on each row.
+message EnrichedResponse {
+  oneof result {
+    // The result data.
+    EnrichedResultSet result_set = 1;
+    // How the query changed the CQL schema.
+    SchemaChange schema_change = 4;
+  }
+  // The server-side warnings for the query, if any.
+  repeated string warnings = 2;
+  // The tracing information, if it was requested for the query.
+  Traces traces = 3;
+}
+
+// The data returned in response to a CQL query, with added data on each row.
+message EnrichedResultSet {
```

Comment:
This would be the way I initially approached this - and I had figured that since they are optional by default that it's a backwards-compatible/non-breaking change. Now that you've said this, I'm not sure what the impetus is for not changing the public API here - by all technicality a current user would be able to use the API in the exact same way even with the added fields.

I can revert this to a previous commit that had everything working by adding two byte values to the Row message

Reply:
but @ivansenic @jeffreyscarpenter what was the reasoning behind not changing the grpc API at all?

Reply:
In the original PR the public API was changed without introducing the enriched parameter in the query. In addition, I am not sure how would you anyway chabge the public api without breaking things, the Response contains the array of rows and here we need the EnrichedRow which contains the two byte params. But maybe I am wrong.

Anyway I am up for it if it's doable. However, I am not sure if this is more work than having a separated enriched stuff on the bridge.

Both options are fine with me.

Reply:
The whole idea of the bridge was that it was separate from the public gRPC API. In my opinion our current situation in which the bridge is reusing protos from the public gRPC API is an undesirable coupling. It leads to situations like this in which we want to add something to the bridge, and then we have to consider the effects on the public gRPC. Why can't we make a clean break and allow these to develop independently?